src/Npgsql/NpgsqlTransaction.cs

```diff
@@ -101,31 +101,35 @@ internal NpgsqlTransaction(NpgsqlConnection conn, IsolationLevel isolationLevel
             _connector.Transaction = this;
             _connector.TransactionStatus = TransactionStatus.Pending;
 
-            switch (isolationLevel) {
-                case IsolationLevel.RepeatableRead:
-                case IsolationLevel.Snapshot:
-                    _connector.PrependInternalMessage(PregeneratedMessage.BeginTrans);
-                    _connector.PrependInternalMessage(PregeneratedMessage.SetTransRepeatableRead);
-                    break;
-                case IsolationLevel.Serializable:
-                    _connector.PrependInternalMessage(PregeneratedMessage.BeginTrans);
-                    _connector.PrependInternalMessage(PregeneratedMessage.SetTransSerializable);
-                    break;
-                case IsolationLevel.ReadUncommitted:
-                    // PG doesn't really support ReadUncommitted, it's the same as ReadCommitted. But we still
-                    // send as if.
-                    _connector.PrependInternalMessage(PregeneratedMessage.BeginTrans);
-                    _connector.PrependInternalMessage(PregeneratedMessage.SetTransReadUncommitted);
-                    break;
-                case IsolationLevel.ReadCommitted:
-                    _connector.PrependInternalMessage(PregeneratedMessage.BeginTrans);
-                    _connector.PrependInternalMessage(PregeneratedMessage.SetTransReadCommitted);
-                    break;
-                case IsolationLevel.Unspecified:
-                    isolationLevel = DefaultIsolationLevel;
-                    goto case DefaultIsolationLevel;
-                default:
-                    throw new NotSupportedException("Isolation level not supported: " + isolationLevel);
+            if (!_connector.IsCrateDB)
```

Comment:
This code appears to silently ignore code calling BeginTransaction(). As a general rule, Npgsql doesn't attempt to catch errors that would be raised by PostgreSQL in any case - it simply relies on PostgreSQL for that. So part of me wants to say that this shouldn't exist: if the user attempts to begin a transaction, CrateDB would presumably return an error that would be raised as a PostgresException.

However, I do understand that a silent ignore may be more appropriate here than an error, since some layers (e.g. Entity Framework) may automatically start transactions in some cases, and we want to avoid errors for that. In that case, it would be preferable to move this check to `NpgsqlConnection.BeginTransaction()` and exit immediately. The code as it's currently written sets various state as if a transaction has been started, although that's not the case.

Also, rather than checking IsCrateDB, please create a SupportsTransactions feature on the connector and check that.